pycaqtimage/pycaqtimage.sip

```diff
@@ -0,0 +1,727 @@
+	    for (int iCol = 0; iCol < imageBuffer->imgwidth; ++iCol) {
+		*dstF++ = *src;
+		*dst++ = *src;
+		src += col_inc;
+	    }
+	    src += row_inc;
+	}
+    } else {
+	for (int iRow = 0; iRow < imageBuffer->imgheight; ++iRow) {
+	    for (int iCol = 0; iCol < imageBuffer->imgwidth; ++iCol) {
+		*dstF += (*src - *dstF) / iNewAverage;
+		*dst++ = *dstF++;
+		src += col_inc;
+	    }
+	    src += row_inc;
+	}
+    }
+    imageBuffer->iNumAveraged = iNewAverage % imageBuffer->iAverage;
+
+    if (iNewAverage == imageBuffer->iAverage)
+	_pyCopyToQImage(imageBuffer, 1);
+}
+
+static void _pyColorImagePvCallback(void* cadata, long count, size_t size, void* usr)
+{
+    ImageBuffer*  imageBuffer = reinterpret_cast<ImageBuffer*>(usr);
+    uint8_t* cadata8 = reinterpret_cast<uint8_t*>(cadata); /* R, G, B! */
+    uint8_t*  src    = cadata8;
+    uint32_t *dst    = imageBuffer->imageData;
+    float    *dstF   = imageBuffer->imageDataF;
+    int orientation  = imageBuffer->orientation;
+    int init_offset  = imageBuffer->size * initMult1[orientation] + 
+	               imageBuffer->srcwidth * initMult2[orientation] +
+                       initK[orientation];
+    int row_inc      = imageBuffer->size * rowIncMult1[orientation] +
+                       imageBuffer->srcwidth * rowIncMult2[orientation] +
+                       rowIncK[orientation];
+    int col_inc      = imageBuffer->srcwidth * colIncMult[orientation] +
+                       colIncK[orientation];
+    int iNewAverage  = imageBuffer->iNumAveraged + 1;
+
+    UNUSED(count);
+    UNUSED(size);
+
+    /* If we're using the color image, don't average, just copy and we're done! */
+    if (!imageBuffer->useGray) {
+        src += 3 * init_offset;
+	for (int iRow = 0; iRow < imageBuffer->imgheight; ++iRow) {
+	    for (int iCol = 0; iCol < imageBuffer->imgwidth; ++iCol) {
+		*dst++ = RGB(src);
+		src += 3 * col_inc;
+	    }
+	    src += 3 * row_inc;
+	}
+	_pyCopyToQImage(imageBuffer, 0);
+    } else {
+	if (iNewAverage == 1) {
+	    for (int iRow = 0; iRow < imageBuffer->imgheight; ++iRow) {
+		for (int iCol = 0; iCol < imageBuffer->imgwidth; ++iCol) {
+		    *dstF = GRAY(src);
+		    *dst++ = *dstF++;
+		    src += 3 * col_inc;
+		}
+		src += 3 * row_inc;
+	    }
+	} else {
+	    for (int iRow = 0; iRow < imageBuffer->imgheight; ++iRow) {
+		for (int iCol = 0; iCol < imageBuffer->imgwidth; ++iCol) {
+		    *dstF += (GRAY(src) - *dstF) / iNewAverage;
+		    *dst++ = *dstF++;
+		    src += 3 * col_inc;
+		}
+		src += 3 * row_inc;
+	    }
+	}
+	imageBuffer->iNumAveraged = iNewAverage % imageBuffer->iAverage;
+
+	if (iNewAverage == imageBuffer->iAverage)
+	    _pyCopyToQImage(imageBuffer, 1);
+    }
+}
+ 
+static void _pyImagePvCallback(void* cadata, long count, size_t size, void* usr)
+{
+  ImageBuffer*  imageBuffer = reinterpret_cast<ImageBuffer*>(usr);
+
+  UNUSED(count);
+  if (size != 1 && size != 2 && size != 4) {
+    printf("Image pixel size is %d bytes?\n", (int) size);
```

Comment:
Seems like this if clause could be handled by `default:` in the switch below

Reply:
True enough.  Fixed.